xbmc/platform/darwin/tvos/TVOSTopShelf.mm

```diff
@@ -46,119 +48,135 @@
   return sTopShelf;
 }
 
-void CTVOSTopShelf::SetTopShelfItems(CFileItemList& movies, CFileItemList& tv)
+void CTVOSTopShelf::SetTopShelfItems(CFileItemList& items, TVOSTopShelfItemsCategory category)
 {
   @autoreleasepool
   {
+    // Retrieve store URL
     auto storeUrl = [tvosShared getSharedURL];
     if (!storeUrl)
       return;
-
     storeUrl = [storeUrl URLByAppendingPathComponent:@"RA" isDirectory:YES];
-    const BOOL isJailbroken = [tvosShared isJailbroken];
-    CLog::Log(LOGDEBUG, "TopShelf: using shared path {} (jailbroken: {})", storeUrl.path.UTF8String,
-              isJailbroken ? "yes" : "no");
 
-    auto sharedDefaults = [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]];
-    auto sharedDictJailbreak = isJailbroken ? [[NSMutableDictionary alloc] initWithCapacity:2 + 2]
-                                            : nil; // for jailbroken devices
+    const BOOL isSandboxed = CDarwinEmbedUtils::IsIosSandboxed();
+    CLog::Log(LOGDEBUG, "[TopShelf] (sandboxed: {}) Use storeUrl: {}", isSandboxed ? "yes" : "no",
+              storeUrl.path.UTF8String);
 
-    // store all old thumbs in array
+    // Store all old thumbs in array
     auto fileManager = NSFileManager.defaultManager;
     auto filePaths =
         [NSMutableSet setWithArray:[fileManager contentsOfDirectoryAtPath:storeUrl.path error:nil]];
     std::string raPath = storeUrl.path.UTF8String;
-    CVideoThumbLoader thumbLoader;
 
+    // Shared dicts  (if we are sandboxed we use sharedDefaults, else we use sharedJailbreak)
+    auto sharedDefaults =
+        isSandboxed ? [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]] : nil;
+    auto sharedJailbreak = isSandboxed ? nil : [[NSMutableDictionary alloc] initWithCapacity:2 + 2];
```

Comment:
Why does it have capacity `2+2`? I see only 2 `fillSharedDicts` calls now.

Reply:
I don't know, it's just the old code that I didn't touch :-/